### ISSUE TITLE
Need better explanation of TMOS versions by CFT version in the matrix

### ISSUE THREAD

**Issue by C0missar** [F5Networks/f5-aws-cloudformation#97]
## Do you already have an issue opened with F5 support?
Github Issues are consistently monitored by F5 staff, but should be considered as best effort only and you should not expect to receive the same level of response as provided by F5 Support. Please [open an case](https://support.f5.com/csp/article/K2633) with F5 if this is a critical issue.

## Description
Describe the problem you're having or the enhancement you'd like to request.

The matrix lists a TMOS version next to each CFT version.  However, the matrix was last updated for v5.0.2, and regardless of version, it appears that the AMIs have all been updated to v15.x - even the 14.1.4 template.  Without going to a custom AMI, the only AMI option available in the CFT is All or LTM only.

What is the meaning of TMOS version listed in the matrix?

## Template
For bugs, enter the template with which you are experiencing issues below.

## Severity Level
For bugs, enter the bug severity level. Do not set any labels.

Severity: <Fill in level: 1 through 5>

Severity level definitions:
 1. **Severity 1 (Critical)** : Defect is causing systems to be offline and/or nonfunctional. immediate attention is required.
 2. **Severity 2 (High)** : Defect is causing major obstruction of system operations.
 3. **Severity 3 (Medium)** : Defect is causing intermittent errors in system operations.
 4. **Severity 4 (Low)** : Defect is causing infrequent interuptions in system operations.
 5. **Severity 5 (Trival)** : Defect is not causing any interuptions to system operations, but none-the-less is a bug.

**Comment by chen23** [comment]
@C0missar I believe that you can match the release of the CFT (i.e. 5.4.0) to the version of BIG-IP.  If you need to use a specific version of BIG-IP (i.e. 13.1) you can use the matrix to go back in time to the release that you need (v4.1.4) the minor version seems to increment when we update the list of AMIs for a particular release.  That is how it works AFAIK.  Something that I find puzzling is why 15.0.1 is listed for all the standalone versions.  Not sure whether that is a typo or accurate.

**Comment by C0missar** [comment]
That may have been the intent, but it is not what is actually deployed.  I am getting code versions later than what is shown in the matrix, regardless of template version.

Regards,

Stan

**Comment by JeffGiroux** [comment]
It's always recommended to use the latest templates as that is where the latest fixes are added. All templates point to the most current releases.

That being said, there's a "customImageId" field where you can supply your own image ID (aka AMI id). So, use the latest template version (currently 5.4.0) and enter a custom AMI image ID during deployment. You can find the relevant template versions and their corresponding images here if you want to use the image that goes along with the template.

https://github.com/F5Networks/f5-aws-cloudformation/blob/master/aws-bigip-version-matrix.md

But...you have the option to supply a custom image ID AND use the latest template. To find a list of available AMI BIG-IP images...
1. Open AWS
2. Click EC2
3. Click AMIs
4. Change dropdown search field to "Public Images"
5. Search for "bigip"
6. Find the version you want and image type you want (ex. 14.1.0.3, 2 boot slots, All modules) = AMI ID  ami-05b0de156bf3306b2.

Deploy the latest version template, enter ami-05b0de156bf3306b2 as custom image ID, report back with your results.

**Comment by shyawnkarim** [comment]
Thanks for reaching out to us about our documentation. I've created internal issue ID ESECLDTPLT-1973 to address this.

**Comment by C0missar** [comment]
I don't really understand that.  The CFT matrix implies a specific TMOS version; changing the AMIs means automated deployments aren't repeatable or idempotent.  Also, most shops only want to deploy TMOS versions that have  been tested in their environments first.

**Comment by C0missar** [comment]
I would still like an explanation of the CFT matrix.  If all CFTs automatically deploy the most recent code, what is the point of the matrix?  This directly contradicts the other statement made in the comment:
```
You can find the relevant template versions and their corresponding images here if you want to use the image that goes along with the template.
```
Why would anybody want to run a template when they had no idea what version of code would be installed?

**Comment by shyawnkarim** [comment]
The CFT Matrix was wrong. All of the current templates deploy the latest 15.0.x image. I've filed a documentation bug to get this updated.

**Comment by C0missar** [comment]
Then what is the purpose of the CFT matrix?  Every entry should be 15.0.x.

**Comment by shyawnkarim** [comment]
@C0missar, the matrix should show what the latest version of BIG-IP was at the time of each template release.

**Comment by C0missar** [comment]
But that ISN'T what the CFT will actually install, which makes the matrix fairly pointless, especially since all the references say to see the matrix for TMOS versions.

**Comment by JeffGiroux** [comment]
Perfect timing. Just released...
https://devcentral.f5.com/s/articles/Using-Cloud-Templates-to-Change-BIG-IP-Versions-AWS

The matrix shows the exact versions that the template is coded to deploy. You can of course change the version through various methods like customImageId. However, the matrix shows the coded versions or image IDs that the template will deploy. Preferrably, the matrix ALSO has an extra column for "Supported BIG-IP Versions" in contrast to the "Templated Coded BIG-IP Versions" (or similar name).

**Comment by shyawnkarim** [comment]
Closing. These legacy templates are now in maintenance mode and are being replaced by our next-generation templates available in the [Cloud Templates 2.0 GitHub repo](https://github.com/F5Networks/f5-aws-cloudformation-v2).

The documentation has been completely rebuilt for V2.
